packages/blockly-field-slider/src/field_slider.js

```diff
@@ -0,0 +1,215 @@
+
+    this.boundEvents_.push(Blockly.bindEventWithChecks_(
+        sliderInput, 'input', this, this.onSliderChange_));
+
+    return wrapper;
+  }
+
+  /**
+   * Disposes of events belonging to the slider editor.
+   * @private
+   */
+  dropdownDispose_() {
+    for (let event of this.boundEvents_) {
+      Blockly.unbindEvent_(event);
+    }
+    this.sliderInput_ = null;
+  }
+
+  /**
+   * Updates the slider when the field rerenders.
+   * @protected
+   * @override
+   */
+  render_() {
+    super.render_();
+    this.updateSlider_();
+  }
+
+  /**
+   * Sets the text to match the slider's position.
+   * @private
+   */
+  onSliderChange_() {
+    this.setEditorValue_(this.sliderInput_.value);
+  }
+
+  /**
+   * Updates the slider when the field rerenders.
+   * @private
+   */
+  updateSlider_() {
+    if (!this.sliderInput_) {
+      return;
+    }
+    this.sliderInput_.setAttribute('value', this.getValue());
+  };
+
+}
+
+/**
+ * Constructs a FieldSlider from a JSON arg object.
+ * @param {!Object} options A JSON object with options (value, min, max, and
+ *                          precision).
+ * @return {!FieldSlider} The new field instance.
+ * @package
+ * @nocollapse
+ */
+FieldSlider.fromJson = function(options) {
+  return new FieldSlider(options['value'],
+      undefined, undefined, undefined, undefined, options);
+};
+
+Blockly.fieldRegistry.register('field_slider', FieldSlider);
+
+/**
+ * CSS for slider field.
+ */
+Blockly.Css.register([
+    /* eslint-disable indent */
+    `.fieldSlider {
+      appearance: none;
+      -webkit-appearance: none;
+      background: #ddd;
+      border-radius: 5px;
+      height: 5px;
+      opacity: 1;
+      outline: none;
+    }
+    /* Webkit */
+    .fieldSlider::-webkit-slider-thumb {
+      -webkit-appearance: none;
+      background: #fff;
+      border-color: #fff;
+      border-radius: 50%;
+      cursor: pointer;
+      height: 10px;
+      opacity: 1;
+      width: 10px;
+    }
+    /* Firefox */
+    .fieldSlider::-moz-range-thumb {
+      border-color: #fff;
+      border-radius: 50%;
+      background: #fff;
+      cursor: pointer;
+      height: 10px;
+      width: 10px;
+    }
+    .fieldSlider::-moz-focus-outer {
+      /* override the focus border style */
+      border: 0;
+    }
+    /* IE */
+    .fieldSlider::-ms-track {
+      /* IE wont let the thumb overflow the track, so fake it */
+      background: transparent;
+      border-color: transparent;
+      border-width: 3px 0;
+      /* remove default tick marks */
+      color: transparent;
+    }
+    .fieldSlider::-ms-fill-lower  {
+      border-radius: 5px;
+      background: #ddd;
+    }
+    .fieldSlider::-ms-fill-upper  {
+      border-radius: 5px;
+      background: #ddd;
+    }
+    .fieldSlider::-ms-thumb {
+      border-color: #fff;
+      border-radius: 50%;
+      background: #fff;
+      cursor: pointer;
+      height: 10px;
+      width: 10px;
+    }`
+    /* eslint-enable indent */
```

Comment:
Thumb size is too small for me, especially on IOS. I would recommend at least a size of 25px.

Reply:
I've increased it to 20px. 25px made it bigger than the field and seemed too big to me.
However, in future PR could look at changing shape of thumb and messing with size. For now, just don't want to spend too much time tweaking it.

Reply:
Sounds good :)